### PR TITLE
Session mgmt

### PR DESCRIPTION
**Pull request by chibDevika**

### CODE REVIEW

File: src/Components/HomePage.jsx [hunk 1]
```diff
@@ -1,34 +1,106 @@
...
+        .then(() => {
+          sendRequest(moment(start).format('YYYY-MM-DD HH:mm:ss'), moment(end).format('YYYY-MM-DD HH:mm:ss'));
+        })
+        .catch(() => {
+          logout();
+    });
+    } else {
+      logout();
+    }
+  };
```
Comment (shubham-gup) [inline src/Components/HomePage.jsx:52]: Should this be memoized?

File: src/Components/HomePage.jsx [hunk 2]
```diff
@@ -1,34 +1,106 @@
...
+          <Typography variant="h6" margin={1}>Start Date and Time</Typography>
+          <DateTimePickerComponent
+            date={start}
+            updateDate={start => setStartDate(moment(start))} />   
+        </div>
+        <div className="inputs">
+          <Typography variant="h6" my={1} mx={1.5}>End Date and Time</Typography>
+          <DateTimePickerComponent
+            date={end}
+            updateDate={end => setEndDate(moment(end))}
```
Comment (shubham-gup) [inline src/Components/HomePage.jsx:80]: Let's avoid inline methods?

File: src/Components/LandingPage.jsx [hunk 3]
```diff
@@ -1,45 +1,79 @@
...
     axios({
       method: 'post',
-      url: 'employees/employees-data/',
+      url: 'employees/employees/',
       data: {
-        username: response.profileObj.googleId,
-        first_name: response.profileObj.givenName,
-        last_name: response.profileObj.familyName,
-        email: response.profileObj.email,
+        username: "testUser",
```
Comment (shubham-gup) [inline src/Components/LandingPage.jsx:25]: `testUser`?

File: src/Components/LandingPage.jsx [hunk 4]
```diff
@@ -1,45 +1,79 @@
...
+        first_name: response.data['given_name'],
+        last_name: response.data['family_name'],
+        email: response.data['email'],
       },
     }).then((res) => {
-      return res.id;
+      console.log(res);
```
Comment (shubham-gup) [inline src/Components/LandingPage.jsx:31]: Let's not leave console log in code after testing?

File: src/Components/LandingPage.jsx [hunk 5]
```diff
@@ -1,45 +1,79 @@
...
+      },
+    }).then((res) => {
+      localStorage.setItem('access_token', res.data.access_token);
+      localStorage.setItem('refresh_token', res.data.refresh_token);
+      localStorage.setItem('expires_in', res.data.expires_in);
+      localStorage.setItem('timestamp', new Date().getSeconds());
+      getUserDetails();
+      navigate('/dashboard');
+    });
+  };
```
Comment (shubham-gup) [inline src/Components/LandingPage.jsx:67]: Should this be memoized?

**Review by shubham-gup — CHANGES_REQUESTED** [review]

File: src/App.css [hunk 6]
```diff
@@ -26,6 +26,41 @@
   color: rgb(112, 76, 182);
 }
 
+.bodyBox {
+  margin-left: 30%;
+  margin-right: 30%;
```
[the remark this answers is not in the document]
Comment (chibDevika) [reply src/App.css:31]: yes done

File: src/App.css [hunk 7]
```diff
@@ -26,6 +26,41 @@
...
+  font-weight: 700;
+}
+
+.logout {
+  margin-top: 45rem;
+}
+
+.landingPageBox {
+  width: 400;
+  height: 200;
```
[the remark this answers is not in the document]
Comment (chibDevika) [reply src/App.css:44]: yes done

File: src/Components/Calendar.jsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  }, []);
+
+  return (
+    <div>
+      <Calendar
+        localizer={localizer}
+        events={myEvents}
+        startAccessor="start"
+        endAccessor="end"
+        style={{ height: 600 }}
```
[the remark this answers is not in the document]
Comment (chibDevika) [reply src/Components/Calendar.jsx:72]: the component doesn't work if I remove it, it's a property of the component I think

File: src/Components/Calendar.jsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    updateCalendar();
+  }, []);
+
+  const logout = () => {
+    localStorage.clear();
+    navigate('/');
+  };
+
+  const updateCalendar = () => {
+    if (localStorage.getItem('access_token')) {
```
[the remark this answers is not in the document]
Comment (chibDevika) [reply src/Components/Calendar.jsx:25]: yes, fixed it

File: src/Components/LandingPage.jsx (hunk 4 above)
Comment (chibDevika) [reply src/Components/LandingPage.jsx:31]: removed it

File: src/Components/LandingPage.jsx (hunk 5 above)
Comment (chibDevika) [reply src/Components/LandingPage.jsx:67]: I had missed it. done now

File: src/Components/LandingPage.jsx (hunk 3 above)
Comment (chibDevika) [reply src/Components/LandingPage.jsx:25]: I had missed it. changed it

File: src/Components/HomePage.jsx (hunk 1 above)
Comment (chibDevika) [reply src/Components/HomePage.jsx:52]: done

File: src/Components/HomePage.jsx (hunk 2 above)
Comment (chibDevika) [reply src/Components/HomePage.jsx:80]: I've changed most of them, but some changes won't come in effect from css file.

File: src/App.css [hunk 10]
```diff
@@ -26,6 +26,47 @@
...
+}
+
+.inputs {
+  display: flex;
+  align-items: center;
+  margin: 0.5rem;
+}
+
+.navBar {
+  background-color: #e6e6e6;
```
Comment (shubham-gup) [inline src/App.css:56]: Should we create constants for colors?

File: src/Components/Calendar.jsx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      response
+        .then(() => {
+          getEvents();
+        }).catch(() => {
+          logout();
+        });
+    } else {
+      logout();
+    }
+  };
```
Comment (shubham-gup) [inline src/Components/Calendar.jsx:37]: Should we warp these with `useCallback`?

File: src/Components/Calendar.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (shubham-gup) [reply src/Components/Calendar.jsx:76]: Can we do it by a class?

**Review by shubham-gup — CHANGES_REQUESTED** [review]

File: src/Components/Calendar.jsx (hunk 11 above)
Comment (chibDevika) [reply src/Components/Calendar.jsx:37]: done

File: src/Components/Calendar.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (chibDevika) [reply src/Components/Calendar.jsx:76]: It didn't work, I tried that

File: src/App.css (hunk 10 above)
Comment (chibDevika) [reply src/App.css:56]: done

**Pull request merged**